### PR TITLE
[DEPRECATION] #14754 deprecate `canDispatchToEventManager`

### PR DESCRIPTION
**Pull request by habdelra**
This is to address the deprecation in #14754.  I'm not sure if the deprecation message provides enough detail.

### CODE REVIEW

File: packages/ember-glimmer/tests/integration/event-dispatcher-test.js [hunk 1]
```diff
@@ -138,6 +139,36 @@ moduleFor('EventDispatcher#setup', class extends RenderingTest {
...
+            assert.ok(true, 'custom event was triggered');
+          }
+        }
+      }),
+      template: `<p>Hello!</p>`
+    });
+
+    expectDeprecation(() => {
+      this.render(`{{x-foo}}`);
+    }, '[DEPERECATED] `canDispatchToEventManager` has been deprecated.');
```
Comment (jpadilla) [inline packages/ember-glimmer/tests/integration/event-dispatcher-test.js]: Should be `[DEPRECATED]` instead of `[DEPERECATED]`.

File: packages/ember-glimmer/tests/integration/event-dispatcher-test.js [hunk 2]
```diff
@@ -138,6 +139,36 @@ moduleFor('EventDispatcher#setup', class extends RenderingTest {
     this.$('div').trigger('myevent');
   }
 
+  ['@test canDispatchToEventManager is deprecated'](assert) {
+    this.dispatcher.canDispatchToEventManager = null;
+    this.registerComponent('x-foo', {
+      ComponentClass: Component.extend({
+        eventManager: {
+          myEvent() {
+            assert.ok(true, 'custom event was triggered');
```
Comment (mmun) [inline packages/ember-glimmer/tests/integration/event-dispatcher-test.js]: It's not clear from how this test is written that this line of code is actually executing. Can you set a boolean and check it at the end, or use assert.expect?

**Review by mmun — COMMENTED** [review]
Thank you for participating in the contributor's workshop!! :)

File: packages/ember-glimmer/tests/integration/event-dispatcher-test.js (hunk 2 above)
Comment (habdelra) [inline packages/ember-glimmer/tests/integration/event-dispatcher-test.js]: whoops--thats copy-pasta, that assert is completely unrelated to this test

**Comment by habdelra** [comment]
@rwjblue could I get your

File: packages/ember-glimmer/tests/integration/event-dispatcher-test.js [hunk 3]
```diff
@@ -138,6 +139,34 @@ moduleFor('EventDispatcher#setup', class extends RenderingTest {
     this.$('div').trigger('myevent');
   }
 
+  ['@test canDispatchToEventManager is deprecated'](assert) {
+    this.dispatcher.canDispatchToEventManager = null;
```
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/event-dispatcher-test.js]: In theory, we should not need to force this to be `null` in this test, the default of `undefined` _should_ be enough

File: packages/ember-views/lib/mixins/view_support.js [hunk 4]
```diff
@@ -538,7 +538,16 @@ export default Mixin.create({
       let owner = getOwner(this);
       let dispatcher = owner && owner.lookup('event_dispatcher:main');
 
-      if (dispatcher && dispatcher.canDispatchToEventManager === null) {
+      deprecate(
+        `[DEPRECATED] \`canDispatchToEventManager\` has been deprecated.`,
```
Comment (rwjblue) [inline packages/ember-views/lib/mixins/view_support.js]: Let's have this message mention `eventManager` instead of `canDispatchToEventManager`, and also add the ${this} to the interpolation (so it's clear to someone seeing the deprecation which component caused it.

File: packages/ember-views/lib/mixins/view_support.js [hunk 5]
```diff
@@ -538,7 +538,16 @@ export default Mixin.create({
...
+        !('canDispatchToEventManager' in dispatcher),
+        {
+          id: 'ember-views.event-dispatcher.canDispatchToEventManager',
+          until: '3.0.0'
```
Comment (rwjblue) [inline packages/ember-views/lib/mixins/view_support.js]: Let's change this to 2.16.0

File: packages/ember-views/lib/system/event_dispatcher.js [hunk 6]
```diff
@@ -129,13 +130,22 @@ export default EmberObject.extend({
...
     @private
   */
-  canDispatchToEventManager: null,
 
   init() {
     this._super();
     assert('EventDispatcher should never be instantiated in fastboot mode. Please report this as an Ember bug.', environment.hasDOM);
+
+    deprecate(
+      `[DEPRECATED] \`canDispatchToEventManager\` has been deprecated.`,
```
Comment (rwjblue) [inline packages/ember-views/lib/system/event_dispatcher.js]: Let's mention which file this is in.

Comment (rwjblue) [inline packages/ember-views/lib/system/event_dispatcher.js]: Also, you should not need to prefix the message with `[DEPRECATED]`

**Review by rwjblue — CHANGES_REQUESTED** [review]

**Comment by habdelra** [comment]
@rwjblue good morning sunshine, I need more

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you @habdelra!
